### ISSUE TITLE
Add info about legend_limit to docs

### ISSUE THREAD

**Issue by rchurt** [holoviz/holoviews#4528]
Might be worth mentioning that that the default of `legend_limit` may prevent legends from being shown.

You could add something like this [here](http://holoviews.org/user_guide/Customizing_Plots.html#Legend-customization): "By default, legends with more than 25 items will not be shown. To change this behavior, pass: `.opts(legend_limit=...)`"

On a related note, I think it would be worth changing the default behavior so that if the number of legend items exceeds `legend_limit`, the legend is still shown, but it either runs off the plot or gets truncated after the first 25 items, instead of just not showing it. I was trying to plot a dataset with a large legend that wasn't showing, and I spent a while trying to figure out why even though I had set `show_legend=True` everywhere I could think of.

**Comment by jbednar** [comment]
> even though I had set show_legend=True everywhere I could think of.

Hmm; that gives me an idea.  The current behavior is deliberate so that overlays of very large numbers of plots are still at least visible, but it is indeed mysterious until someone realizes that there is a `legend_limit`.  So maybe we could redefine `show_legend` to be `None` by default, retaining the current behavior, and then users can force it to either `True` (_always_ show the legend, ignoring `legend_limit`), or `False` (_never_ show the legend, even if under the `legend_limit`).  That way people can always force the behavior they want for a given plot using `show_legend` alone, using an option that has an obvious name and function even for people who have no idea there is a `legend_limit`, while still avoiding having the plot overwritten by a too-large legend by default.

**Comment by rchurt** [comment]
Interesting, I would have expected that setting `show_legend=None` by default to have the same behavior as `show_legend=False`, and that even legends shorter than 25 items wouldn't show. But if you're saying that they would, that could be an interesting idea.

Just so I understand, what's the current behavior with overlays that contain too many items? They just show the legend and not the plot, or they show the plot but the legend runs off the page? This is probably personal preference, but I actually prefer it that way, because it's such an obvious error. If I were new to this, I think I'd prefer to see that than no legend, because then I might wonder if I had fed in the data in such a way that HoloViews couldn't interpret the names of the series and therefore couldn't make a legend from them.

Out of curiosity, if there are too many legend items and you pass `legend_cols=True`, does this allow the plot to be displayed? If so, could that default to `True` if there are too many?

**Comment by jbednar** [comment]
True. Here `None` really just means `the default will be computed`, rather than literally "no legend"; it's unfortunate that there's no `Undefined` literal we could use instead of `None`.  You can see the current behavior by just setting `legend_limit` to 1e10 or some other large number; if I recall correctly it's just that the legend covers the entire plot, but it may also cause other problems. I've found the limit particularly useful for automated reports, where the entire plot is unusable if it's entirely obscured, whereas a plot without a legend at least has some utility.  

I'm not sure what `legend_cols` does...

**Comment by rchurt** [comment]
I see, then perhaps a good compromise would be to have the default behavior be to show up to `legend_limit` entries (so up to 25 by default), and if there are more than that, just don't show them (rather than the current behavior of showing none of them if the number of items exceeds `legend_limit`). Then a description of this behavior could be added to the docs, along with the mention that setting `.opts(legend_limit=...)` will change this behavior.
